server/service/orbit.go

```diff
@@ -213,6 +213,15 @@ func (svc *Service) GetOrbitConfig(ctx context.Context) (fleet.OrbitConfig, erro
 			notifs.NeedsMDMMigration = true
 		}
 
+		inSetupAssistant, err := svc.ds.GetHostAwaitingConfiguration(ctx, host.UUID)
```

Comment:
@mna one way I see is to wrap this up in the `ds.GetHostMDM` call. I could add something like 
```sql
	LEFT OUTER JOIN hosts h ON h.id = hm.host_id
	LEFT OUTER JOIN host_mdm_apple_awaiting_configuration hmaac ON h.uuid = hmaac.host_uuid
```

to the query there. wdyt? is it worth the change?

Reply:
Good question. Since this does not already do a JOIN on `hosts` and we'd need to add it for the `host_uuid`, I think it may not be worth it - could possibly make things worse. One thing we could do though is to run that Awaiting Configuration query only if `isConnectedToFleetMDM` is true, as that's the only case where it can be awaiting config (from us).

Reply:
> could possibly make things worse

yeah that's what I was thinking too.

> One thing we could do though is to run that Awaiting Configuration query only if isConnectedToFleetMDM is true,

good idea!